src/datatype/signature.rs

```diff
@@ -67,60 +65,33 @@ impl FromStr for SignatureType {
         match *self {
             SignatureType::Ed25519 => unimplemented!(),
 
             SignatureType::RsaSsaPss => {
-                Command::new("openssl")
-                    .args(&["rsa", "-inform", "PEM", "-outform", "DER"])
-                    .stdin(Stdio::piped())
-                    .stdout(Stdio::piped())
-                    .stderr(Stdio::piped())
-                    .spawn()
-                    .map_err(Error::Io)
-                    .and_then(|mut child| {
-                        if let Some(ref mut stdin) = child.stdin {
-                            stdin.write_all(key)?;
-                            stdin.flush()?;
-                        } else {
-                            return Err(Error::KeySign("no openssl stdin".into()));
-                        }
-                        Ok(child.wait_with_output()?)
-                    })
-                    .and_then(|output| if output.status.success() {
-                        let key_pair = RSAKeyPair::from_der(Input::from(&output.stdout))
-                            .map_err(|err| Error::KeySign(format!("couldn't read keypair: {}", err)))?;
-                        let mut state = RSASigningState::new(Arc::new(key_pair))
-                            .map_err(|err| Error::KeySign(format!("couldn't get signing state: {}", err)))?;
-                        let mut signature = vec![0; state.key_pair().public_modulus_len()];
-                        state.sign(&RSA_PSS_SHA256, &SystemRandom::new(), msg, &mut signature)
-                            .map_err(|err| Error::KeySign(format!("couldn't sign state: {}", err)))?;
-                        Ok(signature)
-                    } else {
-                        Err(Error::KeySign("RsaSsaPss signing failed".into()))
-                    })
+                let pair = RSAKeyPair::from_der(Input::from(der_key))?;
+                let mut state = RSASigningState::new(Arc::new(pair))?;
+                let mut sig = vec![0; state.key_pair().public_modulus_len()];
+                state.sign(&RSA_PSS_SHA256, &SystemRandom::new(), msg, &mut sig)?;
+                Ok(sig)
             }
         }
     }
 
-    pub fn verify_msg(&self, msg: &[u8], key: &[u8], sig: &[u8]) -> bool {
+    pub fn verify_msg(&self, msg: &[u8], der_key: &[u8], sig: &[u8]) -> bool {
         match *self {
-            SignatureType::Ed25519 => ed25519::verify(msg, key, sig),
+            SignatureType::Ed25519 => ed25519::verify(msg, der_key, sig),
 
             SignatureType::RsaSsaPss => {
                 let verify = || -> Result<bool, Error> {
-                    let pub_key = PKey::from_rsa(Rsa::public_key_from_der(key)?)?;
-                    let mut verifier = OpensslVerifier::new(MessageDigest::sha256(), &pub_key)?;
+                    let pub_key = PKey::from_rsa(Rsa::public_key_from_der(der_key)?)?;
+                    let mut verifier = Verifier::new(MessageDigest::sha256(), &pub_key)?;
```

Comment:
Unfortunately not. The TLS wrappers in the HTTP library still use the openssl lib, though perhaps they can be replaced with a pure rust version. I doubt that would be great for the binary size though which is the main thing we want to optimise.